include/deal.II/base/vectorization.h

```diff
@@ -78,6 +79,37 @@ struct EnableIfScalar<VectorizedArray<Number> >
 };
 
 
+
+#ifndef DEAL_II_WITH_CXX11
+// Specify the types for the implemented multiplications explicitly
+
+template <typename Number>
+struct ProductType<Number, VectorizedArray<Number> >
+{
+  typedef VectorizedArray<Number> type;
+};
+
+template <typename Number>
+struct ProductType<VectorizedArray<Number>, Number>
+{
+  typedef VectorizedArray<Number> type;
+};
+
+template<>
+struct ProductType<double, VectorizedArray<float> >
+{
+  typedef VectorizedArray<float> type;
```

Comment:
I see, thanks for the explanation.

@masterleinad -- can you just add a comment in front of these specializations mentioning that this is indeed the correct way multiplication of these types is implemented?